File: WebApplication1/ViewModels/Solve/SolveViewModel.cs
namespace CrossWorldApp.ViewModels.Solve;
using CrossWorldApp.Models;

public class SolveViewModel
{
    public string SolveId { get; set; }
    
    public int CrosswordId { get; set; }
    
    public double MillisecondsElapsed { get; set; }
    
    public List<List<string>> SolveGrid { get; set; }

    public bool IsSolved { get; set; }
    
    public bool IsCoOp { get; set; }
    
    public bool UsedHints { get; set; }
    
    public string TestCrosswordTitle { get; set; }
    
    public string TestCrosswordAuthor { get; set; }
    
    public List<List<string>> TestCrosswordGrid { get; set; }
}

File: WebApplication1/FrontEnd/components/Solve/Models/SolveViewModel.ts
export class SolveViewModel {
    solveId: string;
    crosswordId: number;
    solveGrid: string[][];
    isSolved: boolean;
    isCoOp: boolean;
    millisecondsElapsed: number;
    usedHints: boolean;
    testCrosswordAuthor: string;
    testCrosswordTitle: string;
    testCrosswordGrid: string[][];
    
    constructor() {
        this.solveId = "";
        this.crosswordId = 0;
        this.solveGrid = [];
        this.isSolved = false;
        this.isCoOp = false;
        this.millisecondsElapsed = 0;
        this.usedHints = false;
        this.testCrosswordAuthor = "";
        this.testCrosswordTitle = "";
        this.testCrosswordGrid = [];
    }
    
}
